src/SConscript.client

```diff
@@ -155,7 +155,9 @@ clientSourceBasic = [
     'third_party/murmurhash3/MurmurHash3.cpp',
     ]
 
+# polyfills for platforms that don't have certain system provided functions
 clientSourceTz = [] if (clientEnv['MONGO_HAVE_TIMEGM'] or windows) else ['third_party/tz/timegm.c']
+clientSourceStrnlen = [] if (clientEnv['MONGO_HAVE_STRNLEN']) else ['mongo/platform/strnlen.cpp']
```

Comment:
Rather than making the list of files more complicated to reason about, maybe just always compile strnlen.cpp, but wrap the entire body in #ifndef MONGO_HAVE_STRNLEN?